Gruntfile.js

```diff
@@ -84,28 +104,47 @@ function getConfig(file, bundleStrategy = SINGLE_BUNDLE, mode = PRODUCTION_MODE)
         export: 'default',
       },
       globalObject: globalObject,
-      publicPath: publicPath,
+
+      // The `publicPath` is the base path for dynamically loaded JS chunks.
+      //   https://webpack.js.org/guides/public-path/
+      //   https://webpack.js.org/configuration/output/#outputpublicpath
+      // It is used by async.ts to import the font files at runtime.
+      // There isn't one setting for `publicPath` that will work for all deployments.
+      // In some scenarios, it needs to be './' to work, but in others it needs to be 'auto' to work.
+      // You can customize the `publicPath` below to work with your production environment.
+      //   publicPath: undefined, // undefined and `auto` are equivalent.
+      //   publicPath: 'auto',    // https://webpack.js.org/guides/public-path/#automatic-publicpath
+      //   publicPath: '',
+      //   publicPath: './',
+      // Our solution:
+      //   Specify the VEX_BASE_PATH environment variable at build time, or
+      //   Specify the VEX_BASE_PATH global variable at runtime.
+      // The value of this option should end in a slash in most cases. For example: VEX_BASE_PATH=/js/
+      publicPath: process.env.VEX_BASE_PATH ?? './',
+      // Or comment out the above line, and set it to whatever you like:
+      // publicPath: '/my/custom/public/path/',
     },
     resolve: {
-      extensions: ['.ts', '...'],
+      extensions: ['.ts', '.tsx', '.js', '...'],
     },
-    devtool: process.env.VEX_GENMAP || mode === PRODUCTION_MODE ? 'source-map' : false,
+    devtool: devtool,
     module: {
       rules: [
         {
-          // Add VERSION and BUILD properties to Vex.Flow.
-          test: /flow\.ts$/,
+          // Add the magic import to async.ts to support dynamic font loading.
+          test: /async\.ts$/,
           loader: 'string-replace-loader',
           options: {
-            multiple: [
-              { search: '_VEX_BUILD_', replace: GIT_COMMIT_HASH },
-              { search: '_VEX_VERSION_', replace: VEXFLOW_VERSION },
-            ],
+            search: '/* IMPORT_WEBPACK_PUBLICPATH_HERE */',
+            replace: "import './webpack_publicpath';",
```

Comment:
We inject an extra import into the CJS builds that load fonts dynamically (e.g., vexflow-core.js, vexflow-core-with-petaluma.js).

The webpack_publicpath.ts file sets the base path for loading the music font chunk files, across various environments including web workers, node, browser.

The ESM build does not need this webpack-specific code.